app/service/validation.py

```diff
@@ -239,4 +239,36 @@ def validate_bulk_import_data(data: dict[str, Any]) -> None:
     if not data:
         raise HTTPException(status_code=status.HTTP_204_NO_CONTENT)
 
+    validate_metadata_values_are_strings(data)
     validate_entity_relationships(data)
+
+
+def _validate_values_are_strings(value: Any) -> None:
+    """
+    Recursively validates that all single values are strings.
+
+    :param Any value: The value to be validated.
+    :raises ValidationError: raised if any single value is not a string.
+    """
+    if isinstance(value, list):
+        for v in value:
+            _validate_values_are_strings(v)
+    elif not isinstance(value, str):
+        raise ValidationError("Metadata values should be strings")
+
+
+def validate_metadata_values_are_strings(data: dict[str, Any]) -> None:
+    """
+    Validates any metadata in the data.
+
+    :param dict[str, Any] data: The data object to be validated.
+    """
+    metadata_values = [
+        value
+        for entity in data.values()
+        if entity is not None
+        for e in entity
+        for value in e.get("metadata", {}).values()
+    ]
+
+    _validate_values_are_strings(metadata_values)
```

Comment:
Instead of writing our own parser, we could use Pydantic's parser? e.g. 

```python
from pydantic import BaseModel, RootModel, ValidationError
from typing import Dict, Union, List

Metadata = RootModel[Dict[str, Union[str, List[str]]]]

if __name__ == "__main__":
    good_json = '{"name": "John", "age": "30", "list": ["item1", "item2"]}'
    bad_json = '{"name": "John", "age": 30, "list": ["item1", "item2"]}'

    try: 
        print(Metadata.model_validate_json(good_json))
        print("Successful") # reaches here
    except ValidationError as e:
        print(e)
        print("Failed")

    try: 
        print(Metadata.model_validate_json(bad_json))
        print("Successful")
    except ValidationError as e:
        print(e)
        print("Failed") # reaches here with a nice descriptive error

```